packages/admin-next/dashboard/src/components/grid/data-grid/data-grid-root/data-grid-root.tsx

```diff
@@ -0,0 +1,627 @@
+        : undefined,
+    overscan: 5,
+  })
+
+  const [anchor, setAnchor] = useState<FieldCoordinates | null>(null)
+
+  const [isSelecting, setIsSelecting] = useState(false)
+  const [selection, setSelection] = useState<FieldCoordinates[]>([])
+
+  const [isDragging, setIsDragging] = useState(false)
+  const [dragSelection, setDragSelection] = useState<FieldCoordinates[]>([])
+
+  const handleFocusInner = (target: HTMLElement) => {
+    const editableField = target.querySelector("[data-field-id]")
+
+    if (editableField instanceof HTMLInputElement) {
+      requestAnimationFrame(() => {
+        editableField.focus()
+        editableField.setSelectionRange(
+          editableField.value.length,
+          editableField.value.length
+        )
+      })
+    }
+  }
+
+  const handleMouseDown = (e: ReactMouseEvent<HTMLTableCellElement>) => {
+    const target = e.target
+
+    /**
+     * Check if the click was on a presentation element.
+     * If so, we don't want to set the anchor.
+     */
+    if (
+      target instanceof HTMLElement &&
+      target.querySelector("[data-role=presentation]")
+    ) {
+      return
+    }
+
+    const rowIndex = parseInt(e.currentTarget.dataset.rowIndex!)
+    const columnIndex = parseInt(e.currentTarget.dataset.columnIndex!)
+
+    const isAnchor = getIsAnchor(rowIndex, columnIndex)
+
+    if (e.detail === 2 || isAnchor) {
+      handleFocusInner(e.currentTarget)
+      return
+    }
+
+    const coordinates: FieldCoordinates = {
+      row: rowIndex,
+      column: columnIndex,
+    }
+
+    setSelection([coordinates])
+    setAnchor(coordinates)
+    setIsSelecting(true)
+  }
+
+  const handleDragDown = (e: ReactMouseEvent<HTMLDivElement>) => {
+    e.stopPropagation()
+    setIsDragging(true)
+  }
+
+  const getIsAnchor = (rowIndex: number, columnIndex: number) => {
+    return anchor?.row === rowIndex && anchor?.column === columnIndex
+  }
+
+  const handleMouseOver = (e: ReactMouseEvent<HTMLTableCellElement>) => {
+    /**
+     * If we're not dragging and not selecting or there is no anchor,
+     * then we don't want to do anything.
+     */
+    if ((!isSelecting && !isDragging) || !anchor) {
+      return
+    }
+
+    const target = e.target
+
+    /**
+     * Check if the click was on a presentation element.
+     * If so, we don't want to add it to the selection.
+     */
+    if (
+      target instanceof HTMLElement &&
+      target.querySelector("[data-role=presentation]")
+    ) {
+      return
+    }
+
+    const rowIndex = parseInt(e.currentTarget.dataset.rowIndex!)
+    const columnIndex = parseInt(e.currentTarget.dataset.columnIndex!)
+
+    /**
+     * If the target column is not the same as the anchor column,
+     * we don't want to add it to the selection.
+     */
+    if (anchor?.column !== columnIndex) {
+      return
+    }
+
+    const direction =
+      rowIndex > anchor.row ? "down" : rowIndex < anchor.row ? "up" : "none"
```

Comment:
suggestion: enum for the directions

Reply:
Personal preference, but I really detest Enums in TS and prefer to use consts 😅